### PR TITLE
Optional auth0

### LINKED ISSUE #96: Make Auth0 an optional dependency
I seems we have quite a lot of users who, understandably, do not want to use Auth0 for their authentication, https://github.com/futurice/pepperoni-app-kit/issues/78, https://github.com/futurice/pepperoni-app-kit/issues/91 among others.

We shipped Pepperoni with the assumption that it would be easier for the users to remove Auth0 than it would be to add it, but this assumption has turned out to be incorrect. Additionally, integrating Auth0 has become a lot easier due to `rnpm` now shipping with React Native.

Let's start work on downgrading Auth0 from a default dependency to an optional one, implemented as sample code and documentation.

### PR DESCRIPTION
**Pull request by chadrien**
closes #96

### CODE REVIEW

File: docs/AUTH0.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+Then link the package to IOS and Android builds with the `link` command:
+
+```bash
+$ react-native link react-native-lock
+```
+
+## Usage
+
+You can see exemple of code used in Pepperoni when Auth0 was still bundled [here](549fd09e).
```
Comment (tino-junge) [inline docs/AUTH0.md]: Link is not working anymore... is it the correct hash you are referring to?

File: docs/AUTH0.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# Auth0
```
Comment (tino-junge) [inline docs/AUTH0.md]: The doc file is not linked currently from the main or setup doc right? Would be helpful though I guess

Comment (chadrien) [inline docs/AUTH0.md]: Ah yeah, correct.

Will add a link.

File: docs/AUTH0.md (hunk 1 above)
Comment (chadrien) [inline docs/AUTH0.md]: Hum… weird. I'm pretty sure I got them directly from github.

¯\\_(ツ)\_/¯ I'll just update them.

File: docs/SETUP.md [hunk 3]
```diff
@@ -93,32 +93,9 @@ More details here: [React Native Android Setup](https://facebook.github.io/react
 
 ### Auth0
 
-#### Configuration
-
-If you don't want to use Auth0, or you want to take it into use later, you can skip this step for now.
-
-1. Before you start you need to create a new application in [Auth0](https://manage.auth0.com/#/applications/)
```
Comment (tino-junge) [inline docs/SETUP.md]: Why are we not moving this part of documentation to the `Auth0.md` file instead of removing it?

**Pull request merged**